src/main/ank/2.3-free-monoids.md

```diff
@@ -0,0 +1,107 @@
+```Haskell
+class Monoid m where
+  mempty :: m
+  mappend :: m -> m -> m
+```
+```kotlin:ank:silent
+interface Monoid<A> {
+  fun empty(): A
+  fun combine(a: A, b: A): A
+}
+```
+................
+```Haskell
+instance Monoid [a] where
+  mempty = []
+  mappend = (++)
+```
+```kotlin
+import arrow.higherkind
```

Comment:
This is ideally, what you're trying to do
`
@extension
interface Monoid<A>: MonoidOf<List<A>>{
  fun empty() = emptyList()
  fun List<A>.combine(other: List<A>) = this + other
}
`

Reply:
These are the imports
```
import arrow.extension
import arrow.typeclasses.MonoidOf
```